### ISSUE TITLE
Scaling doesn't seem to trigger on .net 6 dotnet-isolated function app

### ISSUE THREAD

**Issue by hallgeir-osterbo-visma** [Azure/azure-functions-dotnet-worker#1261]
We've been struggling with getting our Azure Function apps to scale up. It's a function app consisting of an HTTP trigger, running on Elastic Premium. To test, I've created a function that just does a lot of work, and calling that from multiple threads. Maximum burst instances is set to max.

Here's what I've found so far:
- If I deploy a function app that does NOT run dotnet-isolated, using the Microsoft.NET.Sdk.Functions NuGet, then the app scales up just fine.
- If I deploy a function app that have "DotNet-Isolated|6" set in `siteConfig/windowsFxVersion` in the ARM template (this property is set automatically with Terraform if deployed with use_dotnet_isolated_runtime=true), then the function app doesn't scale beyond a single instance. It doesn't seem to make a difference if FUNCTIONS_WORKER_RUNTIME is set to dotnet-isolated (which happens automatically if deploying from Visual Studio), or just leave that setting out.
- For some reason, if I create the function app through the Azure Portal, then the `siteConfig/windowsFxVersion` property is NOT set. Then I can deploy my dotnet-isolated app, set the appsetting FUNCTIONS_WORKER_RUNTIME="dotnet-isolated", and it seems to scale just fine.

I've tested with the latest version of the NuGets:
- Microsoft.Azure.Functions.Worker version 1.10.0
- Microsoft.Azure.Functions.Worker.Sdk version 1.7.0

I found an issue that looks familiar here: https://github.com/Azure/azure-functions-dotnet-worker/issues/792 however this is closed because apparently the solution was to upgrade the Microsoft.Azure.Functions.Worker.Sdk NuGet package to the latest version which I have done. I've also tested with the specific version mentioned for Microsoft.Azure.Functions.Worker.Sdk (1.2.0) - to no avail.

I do have a support ticket open for this as well, where we have given the function app name and so on if you can/need to check any details. The ticket ID is 2301090050001094. 

Is this a bug? What would the advise be here to get this working properly?

**Comment by fabiocav** [comment]
@hallgeir-osterbo-visma thanks for reporting this.

The behavior you've reported in your second bullet above, when using Terraform, is indeed invalid. I wouldn't expect that to prevent scaling, but setting that value with what you have defined will lead to unexpected behavior (like what we're seeing).

Tagging @kulkarnisonia16 to look at the Terraform behavior and work with the appropriate folks to have this corrected.

**Comment by hallgeir-osterbo-visma** [comment]
Thanks @fabiocav for the update here!

What would the correct value be for the windowsFxVersion attribute? Is the correct value basically blank, like it is when I create a function app manually through the portal? If I can know this, then I can probably work around it with using the azapi terraform provider instead.

Just some more info - If I deploy with the regular "dotnet" runtime (in-process), then everything scales perfectly. In that case, Terraform sets windowsFxVersion to DotNet|6, when use_dotnet_isolated_runtime  is set to false.

In Terraform, this value is controlled by the following block (azurerm provider version 3.37.0):
```
  site_config {
    application_stack {
      dotnet_version              = "6"
      use_dotnet_isolated_runtime = true # true => windowsFxVersion=DotNet-Isolated|6, false => windowsFxVersion=DotNet|6
    }
    
```

**Comment by liliankasem** [comment]
@kulkarnisonia16 have you had a chance to look into this? Can we transfer the issue to your team?

**Comment by hallgeir-osterbo-visma** [comment]
I have an update on this actually. We got some assistance on this from MS support in the ticket. It turned out that apparently, the issue was that when the function app is created, and sees an existing secret/key in its storage account, it will try decrypting it. If that fails, it will store a *.snapshot.json file for that secret, and create a new one. If the function app runtime sees 10 or more of those snapshot.json files, the function host will shut down (whatever that means -- at least it seems to have consequences for scaling). In this case we had been experimenting and recreating the function app several times. 

The solution proposed by support was either to regularly delete these snapshot files, or migrate to using Azure Key Vault for secrets, which doesn't have this issue apparently. We've migrated to key vault, and that seems to have solved the issue.
